lib/publiccloud/azure.pm

```diff
@@ -249,24 +254,29 @@ sub generate_tags {
     my $definition = $self->generate_azure_image_definition();
 
 Generated the Azure Image name from the job settings. If present, it takes the PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION setting.
-If not present it generated the image definition name based on distri, version, flavor and SKU.
+If not present, it generates the image definition name based on distri, version, flavor and SKU.
 
+    input $type: 'undefined/null/empty' to provide full image name; ='base' to get the base-name excluding 'sku'.
 Note: Image definitions needs to be distinct names and can only serve one architecture!
 
 Example: 'SLE-MICRO-5.4-BYOS-AZURE-X86_64-GEN2'
+
+    B<return> a string with the image full name distri-version-flavor-arch-sku, if $type undefined; 
+              otherwise the base-name without '-sku'.
 =cut
 
 sub generate_azure_image_definition {
-    my ($self) = @_;
+    my ($self, $type) = @_;
```

Comment:
```suggestion
    my ($self, $basename) = @_;
```

and it could be 1 or undef

Reply:
ok. I was uncertain too, if to use 0/1, but opted for a defined value because then using `f()` or `f(1)` not so easy to distinguish. But atm range is binary, so no need for a long string.

Reply:
done